backend/app/analysis/proximity.py

```diff
@@ -0,0 +1,88 @@
+import nltk
+from collections import Counter
+from more_itertools import windowed
+
+from ..models import (
+    Document,
+    Gender,
+    Corpus,
+)
+
+
+def run_analysis(corpus_id, gender_ids, word_window):
+    """
+    Generates a dictionary of dictionaries for each `Document` object. Each dictionary maps a `Gender` to a word count
+    of words within a specified window of that `Gender`'s pronouns.
+
+    :param corpus_id: An int representing a `Corpus` instance
+    :param gender_ids: A list of ints representing `Gender` ids
+    :param word_window: An integer describing the number of words to look at of each side of a gendered word
+
+    :return: A dict mapping `Document` ids to a dict mapping strings (`Gender` labels) to a `Counter` instance.
+    """
+    results = {}
+
+    doc_ids = Corpus.objects.filter(pk=corpus_id).values_list('documents__pk', flat=True)
+
+    for key in doc_ids:
+        results[key] = generate_gender_token_counters(
+            Document.objects.values_list('tokenized_text', flat=True).filter(pk=key),
+            gender_ids,
+            word_window
+        )
```

Comment:
Could you remind me what the value is in passing the `text_query` rather than the `tokenized_text` itself? That is: evaluate the `tokenized_text` here in line 28 or so, then pass that in. As long as we perform the lookup inside the loop that starts on line 27, we should achieve the same effect as the lazy-loading, right?

Reply:
@MBJean Ah, that makes much more sense! I had originally though of delaying loading the text until the last minute, but I see that this implementation keeps hitting the database for the same list of strings, all in just one iteration through the `doc_ids`. I've changed it so that we load the text at line 28 or so.